viewer/packages/sector-loader/src/GltfSectorRepository.ts

```diff
@@ -0,0 +1,132 @@
+        instancedMeshes: []
+      });
+    }
+
+    if (sector.levelOfDetail == 0) {
+      return Promise.resolve({
+        modelIdentifier: sector.modelIdentifier,
+        metadata: metadata,
+        levelOfDetail: sector.levelOfDetail,
+        instancedMeshes: [],
+        group: undefined
+      });
+    }
+
+    const sectorByteBuffer = await this._sectorFileProvider.getBinaryFile(
+      sector.modelBaseUrl,
+      metadata.sectorFileName!
+    );
+
+    const group = new AutoDisposeGroup();
+
+    const parsedSectorGeometry = this._gltfSectorParser.parseSector(sectorByteBuffer);
+
+    const materials = this._materialManager.getModelMaterials(sector.modelIdentifier);
+
+    parsedSectorGeometry.forEach(parsedGeometry => {
+      const type = parsedGeometry.type as RevealGeometryCollectionType;
+
+      switch (type) {
+        case RevealGeometryCollectionType.BoxCollection:
+          this.createMesh(group, parsedGeometry.buffer, materials.box);
+          break;
+        case RevealGeometryCollectionType.CircleCollection:
+          this.createMesh(group, parsedGeometry.buffer, materials.circle);
+          break;
+        case RevealGeometryCollectionType.ConeCollection:
+          this.createMesh(group, parsedGeometry.buffer, materials.cone);
+          break;
+        case RevealGeometryCollectionType.EccentricConeCollection:
+          this.createMesh(group, parsedGeometry.buffer, materials.eccentricCone);
+          break;
+        case RevealGeometryCollectionType.EllipsoidSegmentCollection:
+          this.createMesh(group, parsedGeometry.buffer, materials.ellipsoidSegment);
+          break;
+        case RevealGeometryCollectionType.GeneralCylinderCollection:
+          this.createMesh(group, parsedGeometry.buffer, materials.generalCylinder);
+          break;
+        case RevealGeometryCollectionType.GeneralRingCollection:
+          this.createMesh(group, parsedGeometry.buffer, materials.generalRing);
+          break;
+        case RevealGeometryCollectionType.QuadCollection:
+          this.createMesh(group, parsedGeometry.buffer, materials.quad);
+          break;
+        case RevealGeometryCollectionType.TorusSegmentCollection:
+          this.createMesh(group, parsedGeometry.buffer, materials.torusSegment);
+          break;
+        case RevealGeometryCollectionType.TrapeziumCollection:
+          this.createMesh(group, parsedGeometry.buffer, materials.trapezium);
+          break;
+        case RevealGeometryCollectionType.NutCollection:
+          this.createMesh(group, parsedGeometry.buffer, materials.nut);
+          break;
+        case RevealGeometryCollectionType.TriangleMesh:
+          this.createMesh(group, parsedGeometry.buffer, materials.triangleMesh);
+          break;
+        case RevealGeometryCollectionType.InstanceMesh:
+          this.createMesh(group, parsedGeometry.buffer, materials.instancedMesh);
+          break;
+        default:
+          //Assert never
+          break;
```

Comment:
```suggestion
        default:
          assertNever(type);
```
If there are types that are not supported/handled, add a case clause for them:
```
case RevealGeometryCollectionType.UnhandledType1:
case RevealGeometryCollectionType.UnhandledType2:
  throw new Error('Unsupported geometry collection type');
```